lib/pavilion/scriptcomposer.py

```diff
@@ -0,0 +1,697 @@
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._explicit_specification = value
+
+    @property
+    def purge( self ):
+        return self._purge
+
+    @purge.setter
+    def purge( self, value ):
+        """Function to set the purge internal variable."""
+        if not isinstance( value, bool ):
+            error = "Purge specification must be of type 'bool' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._purge = value
+
+    @property
+    def swaps( self ):
+        return self._swaps
+
+    @swaps.setter
+    def swaps( self, value ):
+        """Function to set the swaps internal variable."""
+        if value != None and not isinstance( value, OrderedDict):
+            error = "Swaps specification must be of type 'OrderedDict' and " +\
+                    "not {}.".format( type( value ) )
+            raise TypeError( error )
+        self._swaps = value
+
+    @property
+    def unloads( self ):
+        return self._unloads
+
+    @unloads.setter
+    def unloads( self, value ):
+        """Function to set the unloads internal variable."""
+        if value != None and not isinstance( value, list ):
+            error = "Unloads specification must be of type 'list' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._unloads = value
+
+    @property
+    def loads( self ):
+        return self._loads
+
+    @loads.setter
+    def loads( self, value ):
+        """Function to set the loads internal variable."""
+        if value != None and not isinstance( value, list ):
+            error = "Loads specification must be of type 'list' and not " +\
+                    "{}.".format( type( value ) )
+            raise TypeError( error )
+        self._loads = value
+
+    def __init__( self, explicit_specification=None, purge=False, swaps=None,
+                  unloads=None, loads=None ):
+        """Function to set the modules section for the script.
+        :param list explicit_specification: List of commands to manage the
+                                            modules explicitly.  default = None
+        :param bool purge: Whether or not module purge will work on this
+                           machine. default = False
+        :param OrderedDict swaps: Dictionary of modules to swap.  The current
+                                  module should be the key and the module to
+                                  replace it with should be the value.
+                                  default = None.
+        :param list unloads: List of modules to unload for the script. 
+                             default = None.
+        :param list loads: List of modules to load for the script. 
+                           default = None.
+        """
+        self.explicit_specification = explicit_specification
```

Comment:
We need to talk about how module loads, removes, and swaps are supposed to work.  I think we have different visions, and we need to resolve them.

Reply:
I'm also pretty sure we never want to purge.

Reply:
purge is False by default.  I can see wanting to set up your environment in a particular manner, which can start with a purge.